aspnetcore/tutorials/first-web-api.md

```diff
@@ -129,61 +180,94 @@ Run the app:
-
-JSON similar to the following example is returned:
+* The default browser is launched to `https://localhost:<port>`, where `<port>` is the randomly chosen port number displayed in the output. There's no endpoint at `https://localhost:<port>`, so the browser returns [HTTP 404 Not Found](https://developer.mozilla.org/docs/Web/HTTP/Status/404).
 
+* Append `/weatherforecast` to the URL to test the WeatherForecast API. The browser displays JSON similar to the following example:
+    
 ```json
 [
     {
-        "date": "2019-07-16T19:04:05.7257911-06:00",
+        "date": "2025-07-16",
         "temperatureC": 52,
         "temperatureF": 125,
         "summary": "Mild"
     },
     {
-        "date": "2019-07-17T19:04:05.7258461-06:00",
+        "date": "2025-07-17",
         "temperatureC": 36,
         "temperatureF": 96,
         "summary": "Warm"
     },
     {
-        "date": "2019-07-18T19:04:05.7258467-06:00",
+        "date": "2025-07-18",
         "temperatureC": 39,
         "temperatureF": 102,
         "summary": "Cool"
     },
     {
-        "date": "2019-07-19T19:04:05.7258471-06:00",
+        "date": "2025-07-19",
         "temperatureC": 10,
         "temperatureF": 49,
         "summary": "Bracing"
     },
     {
-        "date": "2019-07-20T19:04:05.7258474-06:00",
+        "date": "2025-07-20",
         "temperatureC": -1,
         "temperatureF": 31,
         "summary": "Chilly"
     }
 ]
 ```
+    
+* After testing the web app in the following instruction, press <kbd>Ctrl</kbd>+<kbd>C</kbd> in the integrated terminal to shut it down.
+
+---
+
+### Test the project
+
+# [Visual Studio](#tab/visual-studio)
+
+This tutorial uses [Endpoints Explorer and .http files](xref:test/http-files#use-endpoints-explorer) to test the API.
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+## Create API testing UI with Swagger
+
+There are many available web API testing tools to choose from, and you can follow this tutorial's introductory API test steps with your own preferred tool.
+
+This tutorial utilizes the .NET package [NSwag.AspNetCore](https://www.nuget.org/packages/NSwag.AspNetCore/), which integrates Swagger tools for generating a testing UI adhering to the OpenAPI specification:
+
+* NSwag: A .NET library that integrates Swagger directly into ASP.NET Core applications, providing middleware and configuration.
+* Swagger: A set of open-source tools such as OpenAPIGenerator and SwaggerUI that generate API testing pages that follow the OpenAPI specification.
+* OpenAPI specification: A document that describes the capabilities of the API, based on the XML and attribute annotations within the controllers and models.
+
+For more information on using OpenAPI and NSwag with ASP.NET, see <xref:tutorials/web-api-help-pages-using-swagger>.
+
+### Install Swagger tooling
+
+* Run the following command:
+
+  ```dotnetcli
+  dotnet add package NSwag.AspNetCore
+  ```
```

Comment:
I think this is where you need `# [Visual Studio Code](#tab/visual-studio-code) ` and `VSC`

Reply:
@Rick-Anderson, NSwag/Swagger setup is only being used for VSC here.   The VSC tab is set currently for the entire section here: ## Create API testing UI with Swagger

That seems right the way the tabs are set here. Maybe I don't understand what you are referring to.